File: Data Structures & Algorithms/longest-consecutive-sequence/submission-1.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        int ans=0;
        set<int>s;
        for(auto i:nums) s.insert(i);
        for(auto i:s){
            int cnt=0;
            while(s.find(i++)!=s.end())
                cnt++;
                ans=max(ans,cnt);           
        }
        return ans;
    }
};

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number[]}
     */
    productExceptSelf(nums) {
        const n=nums.length;
        const l = new Array(n);
        const r = new Array(n);
        l[0]=r[n-1]=1;
        for(let i=1;i<n;i++){
            l[i]=l[i-1]*nums[i-1];
        }
        for(let i=n-2;i>=0;i--){
            r[i]=r[i+1]*nums[i+1];
        }
        for(let i=0;i<n;i++) nums[i]=l[i]*r[i];
        return nums;
    }
}
